server/src/http/broadcaster.rs

```diff
@@ -1,54 +1,70 @@
-use std::{
-    hash::{Hash, Hasher},
-    sync::Arc,
-    time::Duration,
-};
+use std::{hash::Hash, sync::Arc, time::Duration};
 
-use actix_web::{
-    rt::time::interval,
-    web::{Json, Query},
-};
+use actix_web::{rt::time::interval, web::Json};
 use actix_web_lab::{
     sse::{self, Event, Sse},
     util::InfallibleStream,
 };
 use dashmap::DashMap;
 use futures::future;
 use prometheus::{register_int_gauge, IntGauge};
-use serde::Serialize;
 use tokio::sync::mpsc;
 use tokio_stream::wrappers::ReceiverStream;
 use tracing::{debug, warn};
-use unleash_types::client_features::{ClientFeatures, Query as FlagQuery};
+use unleash_types::client_features::{ClientFeatures, Query};
 
 use crate::{
     error::EdgeError,
-    feature_cache::FeatureCache,
-    filters::{filter_client_features, name_prefix_filter, project_filter, FeatureFilterSet},
-    tokens::cache_key,
-    types::{EdgeJsonResult, EdgeResult, EdgeToken, FeatureFilters},
+    feature_cache::{FeatureCache, UpdateType},
+    filters::{filter_client_features, name_prefix_filter, FeatureFilter, FeatureFilterSet},
+    types::{EdgeJsonResult, EdgeResult, EdgeToken},
 };
 
-#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
-struct QueryWrapper {
-    query: FlagQuery,
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct StreamingQuery {
+    pub projects: Vec<String>,
+    pub name_prefix: Option<String>,
+    pub environment: String,
 }
 
-impl Hash for QueryWrapper {
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        serde_json::to_string(&self.query).unwrap().hash(state);
+impl From<StreamingQuery> for Query {
+    fn from(value: StreamingQuery) -> Self {
+        Self {
+            tags: None,
+            name_prefix: value.name_prefix,
+            environment: Some(value.environment),
+            inline_segment_constraints: Some(false),
+            projects: Some(value.projects),
+        }
+    }
+}
```

Comment:
We don't use `inline_segment_constraints` and `tags` in the actual feature resolution. 

I've set `inline_segment_constraints` to `false` because that indicates that the SDK supports resolution and if the SDK supports streaming, I think we can assume that it supports segment resolution.

As for tags: Edge doesn't support tag queries.